### ISSUE TITLE
Why do you need an abs operation in the implementation.

### ISSUE THREAD

**Issue by anonymous-submission1991** [ganguli-lab/Synaptic-Flow#10]
Hi,

Thanks for your great paper and codes. Synflow achieves impressive performance in network pruning.

One misunderstanding from my side is that, your paper calculates the score based on p.grad * p, while the implementation adds an abs function on the score. Could you please detailly explain this? I am really confused.

Best,
Miao

for _, p in self.masked_parameters:
    self.scores[id(p)] = torch.clone(p.grad * p).detach().abs_()
    p.grad.data.zero_()

**Comment by danielkunin** [comment]
Check out the line of the paper: "Second, positivity and conservation motivates the construction of a loss function that
yields positive synaptic saliency scores".  This section and theorem 3 will hopefully provide the explanation you are looking for!
